src/Microsoft.Identity.Client/ApiConfig/Parameters/AcquireTokenInteractiveParameters.cs

```diff
@@ -0,0 +1,57 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Identity.Client.Core;
+
+namespace Microsoft.Identity.Client.ApiConfig.Parameters
+{
+    internal class AcquireTokenInteractiveParameters : IAcquireTokenParameters
+    {
+        public Prompt Prompt { get; set; }
+        public OwnerUiParent UiParent { get; } = new OwnerUiParent();
+        public IEnumerable<string> ExtraScopesToConsent { get; set; } = new List<string>();
+        public bool UseEmbeddedWebView { get; set; }
+        public string LoginHint { get; set; }
+        public IAccount Account { get; set; }
+
+        public void LogParameters(ICoreLogger logger)
+        {
```

Comment:
cool. this will be helpful w/customers.

Reply:
see suggestion for adding it every where, perhaps as a requirement on the base interface?

---
In reply to: [248925919](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/798#discussion_r248925919) [](ancestors = 248925919)